### PR TITLE
Add Jump-To hotkeys

### PR DESCRIPTION
**Pull request by dbolack-ab**
This binds CTRL/META-SHIFT-LEFTARROW to brewJump and CTRL/META-SHIFT-RIGHTARROW to sourceJump.

Solves part one in https://github.com/naturalcrit/homebrewery/issues/241

edit: Reflect change in hotkey binding.

 - [x] Agree on which key binds to which jump.
 - [x] Verify brewJump Binding on PC
 - [x] Verify SourceJump Binding on PC
 - [ ]  Verify brewJump Binding on MAC
 - [ ] Verify SourceJump Binding on MAC
 - [ ] Verify brewJump Binding on a Linux
 - [ ] Verify SourceJump Binding on a Linux

### CODE REVIEW

**Comment by 5e-Cleric** [comment]
Those hotkeys are already in use in chrome
For me they open downloads, and dev tools console respectively.

**Comment by calculuschild** [comment]
Now, we don't *need* to add more hotkeys at all. The original issue only suggested it as a compromise if we couldn't get scroll linking to work, but the other PR takes care of that.

**Comment by dbolack-ab** [comment]
> Now, we don't _need_ to add more hotkeys at all (I would actually prefer we don't). One person in the original issue only suggested it as a lesser alternative if we couldn't get scroll linking to work, but the other PR takes care of that.

As a primarily hot key user, I know *I* would use the keys more than the buttons and live scrolling would annoy the crap out of me.

**Comment by calculuschild** [comment]
Fair enough. Just didn't want to add a feature if nobody actually wanted it.

**Comment by Gazook89** [comment]
I think hotkeys for navigating the page is a good idea. And this counts as navigating the page (it is changing the view). 

I do think we can remove hotkeys for many snippets (and I made most of those hot keys so no one should be offended). The snippet menus themselves should have better keyboard navigation (cough radix cough), but in the meantime if there is a snippet hot key that would be better used for something else, I think it’s fine to claim it. 

I intend to work on snippet menu very soon, 1) to convert to function components and 2) look at keyboard navigation and/or radix.

**Comment by 5e-Cleric** [comment]
I'm sorry but:

https://homebrewery.naturalcrit.com/share/1Em9UpHcoKws3xv65lWCRGzPDr3xPAWzvp9-Fp6cxI0Ac

Ctrl-M is used to input spans

File: client/homebrew/editor/editor.jsx [hunk 1]
```diff
@@ -79,6 +85,19 @@ const Editor = createClass({
 		};
 	},
 
+	handleControlKeys : function(e){
+		if(!(e.ctrlKey || e.metaKey)) return;
+		console.log(e);
+		const X_KEY = 88;
+		if((!e.shiftKey) && (e.keyCode == X_KEY)) this.brewJump();
```
Comment (calculuschild) [inline client/homebrew/editor/editor.jsx:92]: `CTRL + x` is "cut". We definitely don't want to overwrite that.

Comment (G-Ambatte) [reply client/homebrew/editor/editor.jsx:92]: As a suggestion:
`CTRL + ALT + RIGHT` => move brew preview to match editor position
`CTRL+ ALT + LEFT` => move editor to match brew preview position

I don't believe these combinations currently have any existing use, please let me know if I'm mistaken

**Comment by G-Ambatte** [comment]
`CTRL+SHIFT+RIGHT` moves the editor, on the _left_ half of the screen
`CTRL+SHIFT+LEFT` moves the preview, on the _right_ half of the screen

To me, this feels backwards - that it would be more intuitive if the operation were reversed.

---

This is, of course, a matter of opinion.

**Comment by dbolack-ab** [comment]
> `CTRL+SHIFT+RIGHT` moves the editor, on the _left_ half of the screen `CTRL+SHIFT+LEFT` moves the preview, on the _right_ half of the screen
> 
> To me, this feels backwards - that it would be more intuitive if the operation were reversed.

I really don't have a strong opinion one way or the other.

**Comment by dbolack-ab** [comment]
Does anyone have strong feelings over which arrow should impact which direction? 

 @calculuschild @G-Ambatte @ericscheid @5e-Cleric

**Comment by calculuschild** [comment]
I would say the arrow should indicate the direction of info transfer.  I.e. pointing right is sending the position from left to right.

**Comment by Gazook89** [comment]
With the :has() selector, or one of the other newer ones, could we style each pane based on which "scroll arrow" is being hovered?

This is kind of funky idea, but if a user is hovering the "scroll preview to match editor" button, the preview pane would show a faint, but visible, double-sided arrow overlaid on it.  And vice versa.

Quick mockup, for hovering the "scroll preview to match editor":
image

I've spent about 0.5 seconds thinking of the css for this, but it seems like it should be possible (or maybe it's a JS thing).  Maybe it even points which direction the pane will scroll....(but then you could have another argument about which way an arrow should point to indicate the page will scroll "up")

Obviously this doesn't solve the immediate question about which way arrows point, but I think it would make it very clear to the user what is about to happen, both on the 'first try' and every subsequent try.

File: client/homebrew/editor/editor.jsx [hunk 2]
```diff
@@ -74,6 +76,10 @@ const Editor = createClass({
 
 	componentDidUpdate : function(prevProps, prevState, snapshot) {
 		this.highlightCustomMarkdown();
+		if(this.props.jumpSource) {
+			this.sourceJump();
+			this.setState({sourceJump: false});
+		}
```
Comment (calculuschild) [inline client/homebrew/editor/editor.jsx:82]: Is this bit of code actually doing anything? Does `this.props.jumpSource` ever get set? And does `this.state.sourceJump` ever get used?

**Comment by calculuschild** [comment]
Looks good. Gonna merge this.

**Pull request merged**